nixos/modules/services/networking/livekit.nix

```
@@ -0,0 +1,135 @@
+{
+  config,
+  lib,
+  pkgs,
+  utils,
+  ...
+}:
+let
+  cfg = config.services.livekit;
+  format = pkgs.formats.json { };
+in
+{
+  meta.maintainers = with lib.maintainers; [ quadradical ];
+  options.services.livekit = {
+    enable = lib.mkEnableOption "Enable the livekit server";
+    package = lib.mkPackageOption pkgs "livekit" { };
+
+    keyFile = lib.mkOption {
+      type = lib.types.path;
+      description = ''
+        LiveKit key file, with syntax `APIkey: secret`.
+        The key and secret are used by other clients or services to connect to your Livekit instance.
+      '';
```

The syntax part is hard to understand. Just point users to `livekit-server generate-keys` instead.

---

generate-keys doesn't output the same syntax though, so how should the user know what to put?

---

FWIW this is the same syntax that https://github.com/element-hq/lk-jwt-service#configuration talks about. I think it's okay.

---

To hammer this in my brain, the syntax is

```json
APIkey: <secret>
```

with the key being `APIkey` and fixed, and the secret being the variable part. Maybe the angle brackets would make a difference in driving this point home.

---

No, you choose both APIKey and secret, neither are set. It's just `<thing>: <thing at least 30 chars long>`

---

So one is name, and the other is a secret and there can be multiple?

```
lk-jwt-service: dIDT9KxQPoyq42Ef
other-service: fhcAqUYFbLdl3UAp
```

And then I only need to hand the one credential to the other service.

---

```suggestion
      description = ''
        LiveKit key file holding one or multiple application secrets. Use `livekit-server generate-keys` to generate a random key name and secret.
        
        The file should have the following format:
        ```
        # <keyname>: <secret>
        lk-jwt-service: CorrectHorseBatteryStaple
        ```
        
        Individual key/secret pairs need to be passed to clients to connect to this instance.
      '';
```

---

Hi, I'll do this tomorrow.

---

Done.